### ISSUE TITLE
What is the expected behaviour of BehavesAsMap?

### ISSUE THREAD

**Issue by victornoel** [yegor256/cactoos#927]
The behaviour of `BehavesAsMap` is ambiguous.

If I write `new BehavesAsMap("a", "b")`, I am not expecting `b` to be anything else than the value of the key `a`.

But many tests makes this assumption:
```
SyncMapTest#behavesAsMap()
SyncMapTest#worksInThreads()
StickyMapTest#behavesAsMap()
SolidMapTest#behavesAsMap()
SolidMapTest#worksInThreads()
MapOfTest#behavesAsMap()
```

It should mean that `b` is indeed the value of `a` and the tests should be modified to comply with this.

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by llorllale** [comment]
@victornoel which tests don't assume that `b` is the value of `a`?

**Comment by victornoel** [comment]
@llorllale 
```
SyncMapTest#behavesAsMap()
SyncMapTest#worksInThreads()
StickyMapTest#behavesAsMap()
SolidMapTest#behavesAsMap()
SolidMapTest#worksInThreads()
MapOfTest#behavesAsMap()
```

Also note that I edited the description, I won't take care of this in #903, I already worked too much on it :)

**Comment by llorllale** [comment]
@victornoel you're right, the problem is in `BehavesAsMap.matchesSafely`

**Comment by llorllale** [comment]
@victornoel can you rephrase this ticket as a bug ?

**Comment by 0crat** [comment]
Job [#927](https://github.com/yegor256/cactoos/issues/927) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@llorllale I rewrote the description

**Comment by 0crat** [comment]
The job [#927](https://github.com/yegor256/cactoos/issues/927) assigned to @dennis00010011b[/z](https://www.0crat.com/u/dennis00010011b), here is [why](https://www.0crat.com/footprint/C63314D6Z/a6c078ff-e378-496d-a2ca-82ecb048360e); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
The job [#927](https://github.com/yegor256/cactoos/issues/927) assigned to @umbrah[/z](https://www.0crat.com/u/umbrah), here is [why](https://www.0crat.com/footprint/C63314D6Z/39f1cd66-aa56-41d6-b8cc-c36afef14609); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by Umbrah** [comment]
@victornoel Can we close it? PR was merged

**Comment by victornoel** [comment]
@Umbrah thanks

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @umbrah[/z](https://www.0crat.com/u/umbrah), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#927](https://github.com/yegor256/cactoos/issues/927) is now out of scope

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @umbrah[/z](https://www.0crat.com/u/umbrah)
